### PR TITLE
✨ add link

### PR DESCRIPTION
**Pull request by YounesBessa**

### CODE REVIEW

File: examples/next-typescript/pages/link.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        <div
+          style={{
+            display: "flex",
+            gap: "16px",
+            alignItems: "center",
+            justifyContent: "center",
+            flexDirection: "column",
+          }}
+        >
+          <Link href="#" variant="primary">
```
Comment (AntoineKM) [inline examples/next-typescript/pages/link.tsx:78]: don't use "#" put real links instead

**Review by AntoineKM — CHANGES_REQUESTED** [review]

File: packages/kitchen/src/components/Link/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import styled from "styled-components";
+import { KitchenComponent } from "../../types";
+
+export type LinkProps = KitchenComponent & {
```
Comment (jerembdn) [inline packages/kitchen/src/components/Link/index.tsx:5]: Ajouter une prop `disabled`

File: packages/kitchen/src/components/Link/index.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  ({
+    as: Component = "a",
+    href,
+    id,
+    className,
+    onClick,
+    variant,
+    children,
+    ...props
+  }: LinkProps) => {
```
Comment (jerembdn) [inline packages/kitchen/src/components/Link/index.tsx:42]: Si `disabled` est undefined, la définir avec la condition `onClick || href` et la donner au comp

File: packages/kitchen/src/components/Link/index.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+  padding: 8px 12px;
+
+  font-size: ${({ theme }) => theme.size.normal};
+
+  text-decoration: ${({ variant }) =>
+    variant === "highlight" || variant === "blend" ? "underline" : "none"};
+
+    :hover {
+    cursor: ${({ onClick, href }) => (onClick || href ? "pointer" : "text")};
```
Comment (jerembdn) [inline packages/kitchen/src/components/Link/index.tsx:93]: ici détecter uniquement si c'est `disabled` (préalablement définie sur cette condition).

Si c'est disabled : mettre le cursor en normal

**Review by jerembdn — CHANGES_REQUESTED** [review]

File: examples/next-typescript/pages/link.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        <div
+          style={{
+            display: "flex",
+            gap: "16px",
+            alignItems: "center",
+            justifyContent: "center",
+            flexDirection: "column",
+          }}
+        >
+          <Link href="https://tonightpass.com/" variant="primary">
```
Comment (AntoineKM) [inline examples/next-typescript/pages/link.tsx:78]: please use local links, example: ``/``

**Review by AntoineKM — CHANGES_REQUESTED** [review]

File: packages/kitchen/src/components/Link/index.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    cursor: ${({ disabled }) => (disabled ? "default" : "pointer")};
+        color: ${({ theme, variant, disabled }) => {
+          if (variant === "secondary") {
+            return theme.colors.text.light;
+          } else if (disabled) {
+            return theme.colors.text.lighter;
+          } else {
+            return theme.colors.text.lighter;
+          }
+        }};
```
Comment (AntoineKM) [inline packages/kitchen/src/components/Link/index.tsx:110]: missing a closing curly brace for ``:hover {``

File: packages/kitchen/src/components/Link/index.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    } else if (variant === "secondary" || disabled) {
+      return theme.colors.text.light;
+    } else {
+      return theme.colors.text.lightest;
+    }
+  }};
+ 
+  
+
+  padding: 8px 12px;
```
Comment (AntoineKM) [inline packages/kitchen/src/components/Link/index.tsx:93]: padding to a link ????

**Review by AntoineKM — CHANGES_REQUESTED** [review]

File: examples/next-typescript/pages/link.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+          }}
+        >
+          <Link href="/" variant="primary">
+            Primary variant
+          </Link>
+          <Link href="/" variant="secondary">
+            Secondary variant
+          </Link>
+          <Link href="/" variant="highlight">
+            Highlight variant
```
Comment (AntoineKM) [inline examples/next-typescript/pages/link.tsx:85]: our highlight varient is accent.primary

File: packages/kitchen/src/components/Link/index.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    return "transparent";
+  }};
+
+  color: ${({ theme, variant, disabled }) => {
+    if (variant === "highlight") {
+      return theme.colors.accent.info;
+    } else if (variant === "blend") {
+      return theme.colors.accent.danger;
+    } else if (variant === "secondary" || disabled) {
+      return theme.colors.text.light;
```
Comment (AntoineKM) [inline packages/kitchen/src/components/Link/index.tsx:85]: why light and not lighter ?

File: packages/kitchen/src/components/Link/index.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    return theme.colors.layout.dark;
+  }}
+    return "transparent";
+  }};
+
+  color: ${({ theme, variant, disabled }) => {
+    if (variant === "highlight") {
+      return theme.colors.accent.info;
+    } else if (variant === "blend") {
+      return theme.colors.accent.danger;
```
Comment (AntoineKM) [inline packages/kitchen/src/components/Link/index.tsx:83]: go here: https://vercel.com/design/link#blend
- change the type of the note to "success"
- inspect the code to understand what is blend variant

**Review by AntoineKM — CHANGES_REQUESTED** [review]
When you hover a link it should have no effect.

**Comment by AntoineKM** [comment]
@YounesBessa

```
Cloning github.com/tonightpass/kitchen (Branch: younesbessa/link, Commit: 962ef0e)
--
13:45:42.026 | Cloning completed: 308.785ms
13:45:49.348 | Restored build cache
13:45:49.402 | Running "vercel build"
13:45:50.064 | Vercel CLI 28.5.6
13:45:50.534 | Running "install" command: `cd ../../packages/kitchen; yarn link; cd ../../examples/next-typescript; yarn link @tonightpass/kitchen; yarn install`...
13:45:50.908 | yarn link v1.22.17
13:45:50.926 | warning package.json: No license field
13:45:50.945 | warning package.json: No license field
13:45:50.948 | success Registered "@tonightpass/kitchen".
13:45:50.949 | info You can now run `yarn link "@tonightpass/kitchen"` in the projects where you want to use this package and it will be used instead.
13:45:50.949 | Done in 0.05s.
13:45:51.181 | yarn link v1.22.17
13:45:51.220 | success Using linked package for "@tonightpass/kitchen".
13:45:51.220 | Done in 0.05s.
13:45:51.511 | yarn install v1.22.17
13:45:51.629 | [1/4] Resolving packages...
13:45:52.345 | success Already up-to-date.
13:45:52.348 | Done in 0.84s.
13:45:52.364 | Detected Next.js version: 13.0.2
13:45:52.365 | Running "cd ../..; yarn build; cd ./examples/next-typescript; yarn build"
13:45:52.716 | yarn run v1.22.17
13:45:52.756 | $ preconstruct build
13:45:53.444 | 🎁 info building bundles!
13:46:02.897 | 🎁 success built bundles!
13:46:02.931 | Done in 10.22s.
13:46:03.310 | yarn run v1.22.17
13:46:03.350 | $ next build
13:46:04.044 | info  - Linting and checking validity of types...
13:46:09.430 |  
13:46:09.431 | ./pages/_document.tsx
13:46:09.431 | 11:11  Warning: Using `<link />` outside of `<Head>` will disable automatic font optimization. This is discouraged. See: https://nextjs.org/docs/messages/no-page-custom-font  @next/next/no-page-custom-font
13:46:09.432 | 15:11  Warning: Using `<link />` outside of `<Head>` will disable automatic font optimization. This is discouraged. See: https://nextjs.org/docs/messages/no-page-custom-font  @next/next/no-page-custom-font
13:46:09.432 |  
13:46:09.433 | info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
13:46:12.528 | Failed to compile.
13:46:12.529 |  
13:46:12.529 | ./pages/link.tsx:78:26
13:46:12.529 | Type error: No overload matches this call.
13:46:12.530 | Overload 1 of 2, '(props: { slot?: string; style?: CSSProperties; title?: string; children?: ReactNode; as?: ElementType<any>; defaultChecked?: boolean; defaultValue?: string \| number \| readonly string[]; ... 253 more ...; href?: string; } & { ...; } & { ...; }): ReactElement<...>', gave the following error.
13:46:12.531 | Type '"primary"' is not assignable to type '"secondary" \| "highlight" \| "blend"'.
13:46:12.531 | Overload 2 of 2, '(props: StyledComponentPropsWithAs<({ as: Component, href, id, className, disabled, onClick, variant, children, ...props }: LinkProps) => Element, DefaultTheme, { as?: ElementType<any>; children?: ReactNode; } & HTMLAttributes<HTMLElement> & { ...; }, never, ({ as: Component, href, id, className, disabled, onClick, variant, children, ...props }: LinkProps) => Element, ({ as: Component, href, id, className, disabled, onClick, variant, children, ...props }: LinkProps) => Element>): ReactElement<...>', gave the following error.
13:46:12.531 | Type '"primary"' is not assignable to type '"secondary" \| "highlight" \| "blend"'.
13:46:12.531 |  
13:46:12.532 | 76 \|           }}
13:46:12.532 | 77 \|         >
13:46:12.532 | > 78 \|           <Link href="/" variant="primary">
13:46:12.532 | \|                          ^
13:46:12.532 | 79 \|             Primary variant
13:46:12.533 | 80 \|           </Link>
13:46:12.533 | 81 \|           <Link href="/" variant="secondary">
13:46:12.571 | error Command failed with exit code 1.
13:46:12.571 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
13:46:12.598 | Error: Command "cd ../..; yarn build; cd ./examples/next-typescript; yarn build" exited with 1
```

**Review by AntoineKM — CHANGES_REQUESTED** [review]
1. You have completely removed the brightness effect when it is not about secondary,
2. You have totally forgotten to make the secure link part with the reference to @onruntime/next-link (https://next-link-with-typescript.vercel.app/ | https://github.com/onRuntime/next-link)

**Review by AntoineKM — CHANGES_REQUESTED** [review]
image
``#`` It should not open a new window

**Comment by YounesBessa** [comment]
> image `#` should not open a new window

It doesn't look like that anymore, now I replaced this example with the internal and external href examples and now external hrefs open in new tabs but internal hrefs don't (just like you did in https://next-link-with-typescript.vercel.app/ (https://github.com/onRuntime/next-link))

**Review by jerembdn — APPROVED** [review]

**Review by jerembdn — APPROVED** [review]

**Review by AntoineKM — APPROVED** [review]

**Pull request merged**